accounts/abi/bind/backends/simulated.go

```diff
@@ -65,8 +67,11 @@ type SimulatedBackend struct {
 
 // NewSimulatedBackend creates a new binding backend using a simulated blockchain
 // for testing purposes.
-func NewSimulatedBackend(alloc core.GenesisAlloc, gasLimit uint64) *SimulatedBackend {
-	database := rawdb.NewMemoryDatabase()
+func NewSimulatedBackend(database ethdb.Database, alloc core.GenesisAlloc, gasLimit uint64) *SimulatedBackend {
```

Comment:
Hmmm, this will break our API for many people. Could we instead have a new method called `NewSimulatedBackendWithDatabase`? That way you can have the extended functionality, but we don't break APIs + the default tester use case remains simple without needing `nil` parameters to be passed.